lib/pavilion/test_run.py

```diff
@@ -407,6 +402,39 @@ def run_cmd(self):
 
         return '{} run {}'.format(pav_path, self.id)
 
+    def _make_name(self):
+        """Generate the name for this test from the config. This
+        will create a subtitle automatically for permuted tests."""
+
+        parts = [
+            self.config.get('suite', '<unknown>'),
+            self.config.get('name', '<unnamed>'),
+        ]
+
+        subtitle = self.config.get('subtitle')
+        permute_on = self.config.get('permute_on', [])
+        if subtitle is None and permute_on:
+            sub_parts = []
+            var_dict = self.var_man.as_dict()
+            for var in permute_on:
+                if var in var_dict:
+                    val = var_dict[var]
+                    # This can only be a dict or a str.
+                    if isinstance(val, dict):
+                        shortest = ''
+                        for subval in val.values():
+                            if not shortest or len(subval) < len(shortest):
+                                shortest = subval
+                        sub_parts.append(shortest)
+                    else:
+                        sub_parts.append(val)
+            subtitle = '-'.join(sub_parts)
+
+        if subtitle is not None:
```

Comment:
The logic here seems to cover:
- There is no provided subtitle but at least one variable is being permuted over.
- There is a provided subtitle, but no variables being permuted over.

It does not seem to cover if there is a provided subtitle and there is at least one variable being permuted over.

Did I miss that somewhere?

Reply:
Hold off on this one. I don't like how I calculate the key, and I want team input.